RELEASES.md

```diff
@@ -1,3 +1,132 @@
+Version 1.80 (2024-07-25)
+==========================
+
+<a id="1.80-Language"></a>
+
+Language
+--------
+- [Document maximum allocation size](https://github.com/rust-lang/rust/pull/116675/)
+- [Allow zero-byte offsets and ZST read/writes on arbitrary pointers](https://github.com/rust-lang/rust/pull/117329/)
+- [Support C23's variadics without a named parameter](https://github.com/rust-lang/rust/pull/124048/)
+- [Stabilize exclusive_range_pattern](https://github.com/rust-lang/rust/pull/124459/)
+- [Guarantee layout and ABI of `Result` in some scenarios](https://github.com/rust-lang/rust/pull/124870)
+- [Warn (or error) when `Self` ctor from outer item is referenced in inner nested item](https://github.com/rust-lang/rust/pull/124187/)
+- [Turn non-structural-const-in-pattern lints into hard errors](https://github.com/rust-lang/rust/pull/124661/)
+- [Make `WHERE_CLAUSES_OBJECT_SAFETY` lint a regular object safety violation](https://github.com/rust-lang/rust/pull/125380/)
+
+
+<a id="1.80-Compiler"></a>
+
+Compiler
+--------
+- [Update cc crate to v1.0.97 allowing additional specter mitigations on MSVC targets](https://github.com/rust-lang/rust/pull/124892/)
+- [Allow field reordering on types marked `repr(packed(1))`](https://github.com/rust-lang/rust/pull/125360/)
+- [Detect unused structs even when implementing private traits](https://github.com/rust-lang/rust/pull/122382/)
+- [Add a lint against never type fallback affecting unsafe code](https://github.com/rust-lang/rust/pull/123939/)
+- [Add `--print=check-cfg` to get the expected configs](https://github.com/rust-lang/rust/pull/124320/)
+- [Add `rustfmt` cfg to well known cfgs list](https://github.com/rust-lang/rust/pull/124742/)
```

Comment:
while this is a user facing change it doesn't seems worth while mentioning it, I don't think we should mention every change in the well known cfgs list
```suggestion
```